databricks/koalas/tests/test_dataframe.py

```diff
@@ -1750,6 +1751,145 @@ def test_binary_operators(self):
             lambda: ks.range(10).add(ks.range(10).id),
         )
 
+    def test_binary_operator_add(self):
+        # Positive
+        pdf = pd.DataFrame({"a": ["x"], "b": ["y"]})
+        kdf = ks.from_pandas(pdf)
+
+        self.assert_eq(kdf["a"] + kdf["b"], pdf["a"] + pdf["b"])
+
+        # Negative
+        kdf = ks.DataFrame({"a": ["x"], "b": [1]})
+        ks_err_msg = re.escape("string addition can only be applied to string series or literals")
+
+        self.assertRaisesRegex(TypeError, ks_err_msg, lambda: kdf["a"] + kdf["b"])
+        self.assertRaisesRegex(TypeError, ks_err_msg, lambda: kdf["b"] + kdf["a"])
+        self.assertRaisesRegex(TypeError, ks_err_msg, lambda: kdf["b"] + "literal")
+        self.assertRaisesRegex(TypeError, ks_err_msg, lambda: "literal" + kdf["b"])
+        self.assertRaisesRegex(TypeError, ks_err_msg, lambda: 1 + kdf["a"])
```

Comment:
I think we're missing of test for non-string type of `IndexOpsMixin`?

For example.

```python
pdf = pd.DataFrame({"a": ["x"], "b": ["y"], "c": [1]})
kdf = ks.from_pandas(pdf)

...

self.assertRaisesRegex(TypeError, ks_err_msg, lambda: kdf["c"] + "a")
self.assertRaisesRegex(TypeError, ks_err_msg, lambda: kdf["c"] + kdf["a"])
```

or just add

```python
self.assertRaisesRegex(TypeError, ks_err_msg, lambda: ks.Series([1]) + "a")
self.assertRaisesRegex(TypeError, ks_err_msg, lambda: ks.Series([1]) + kdf["a"])
```

Reply:
Good catch! Added missing test cases of non-string type of IndexOpsMixin.